e2e/credentials-server.e2e.spec.ts

```diff
@@ -34,17 +36,33 @@ import { wslHostIPv4Address } from '@/utils/networks';
 import { spawnFile } from '@/utils/childProcess';
 import { findHomeDir } from '@/config/findHomeDir';
 
+let credStore: string;
+let failedEraseMessage: string;
+
 function haveCredentialServerHelper(): boolean {
   // Not using the code from `httpCredentialServer.ts` because we can't use async code at top-level here.
   const dockerConfigPath = path.join(findHomeDir() ?? '', '.docker', 'config.json');
 
   try {
     const contents = JSON.parse(fs.readFileSync(dockerConfigPath).toString());
-    const credStore = contents.credsStore;
+    const credStoreAttempt = contents.credsStore;
 
-    if (!credStore) {
+    if (!credStoreAttempt) {
       return false;
     }
+    credStore = credStoreAttempt;
+
+    switch (credStore) {
+    case 'osxkeychain':
+      failedEraseMessage = 'The specified item could not be found in the keychain';
+      break;
+    case 'wincred':
+      failedEraseMessage = '';
+      break;
+    default:
+      failedEraseMessage = `Need a value for failedEraseMessage for credential helper ${ credStore }`;
+    }
```

Comment:
Rather than looking for a specific error message I think you should just check that you get a non-zero exit status. The error messages are outside our control and subject to change.

Reply:
That is what `curl --fail-with-body` already does:

> --fail-with-body will instead first save the body per normal conventions and then return an error if the HTTP response code was 400 or larger.

Reply:
All those changes to the shell script have been dropped.